src/States/MapViewStateDraw.cpp

```diff
@@ -254,37 +266,12 @@ void MapViewState::drawNavInfo()
 {
 	Renderer& r = Utility<Renderer>::get();
 
-	NAS2D::Rectangle_2d currentIconBounds = MOVE_DOWN_ICON;
-	bool isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	NAS2D::Color iconColor{255, 255, 255, 255};
-	NAS2D::Color iconHighlightColor{255, 0, 0, 255};
-	NAS2D::Color color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 64, 128, 32, 32, color.red(), color.green(), color.blue(), color.alpha());
-
-	currentIconBounds = MOVE_UP_ICON;
-	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 96, 128, 32, 32, color.red(), color.green(), color.blue(), color.alpha());
-
-	currentIconBounds = MOVE_EAST_ICON;
-	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 32, 128, 32, 16, color.red(), color.green(), color.blue(), color.alpha());
-
-	currentIconBounds = MOVE_WEST_ICON;
-	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 32, 144, 32, 16, color.red(), color.green(), color.blue(), color.alpha());
-
-	currentIconBounds = MOVE_NORTH_ICON;
-	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 0, 128, 32, 16, color.red(), color.green(), color.blue(), color.alpha());
-
-	currentIconBounds = MOVE_SOUTH_ICON;
-	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
-	color = isMouseInIcon ? iconHighlightColor : iconColor;
-	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 0, 144, 32, 16, color.red(), color.green(), color.blue(), color.alpha());
+	drawNavIcon(r, MOVE_DOWN_ICON, NAS2D::Rectangle_2d{64, 128, 32, 32}, NAS2D::Color::White, NAS2D::Color::Red);
+	drawNavIcon(r, MOVE_UP_ICON, NAS2D::Rectangle_2d{96, 128, 32, 32}, NAS2D::Color::White, NAS2D::Color::Red);
+	drawNavIcon(r, MOVE_EAST_ICON, NAS2D::Rectangle_2d{32, 128, 32, 16}, NAS2D::Color::White, NAS2D::Color::Red);
+	drawNavIcon(r, MOVE_WEST_ICON, NAS2D::Rectangle_2d{32, 144, 32, 16}, NAS2D::Color::White, NAS2D::Color::Red);
+	drawNavIcon(r, MOVE_NORTH_ICON, NAS2D::Rectangle_2d{0, 128, 32, 16}, NAS2D::Color::White, NAS2D::Color::Red);
+	drawNavIcon(r, MOVE_SOUTH_ICON, NAS2D::Rectangle_2d{0, 144, 32, 16}, NAS2D::Color::White, NAS2D::Color::Red);
```

Comment:
Nice. That really improved the code.

Reply:
I don't know why I didn't do it earlier. I have a personal rule, "If you repeat code more than twice, make it a function." I was clearly repeating code *six* times.